src/core/qgshelp.cpp

```diff
@@ -0,0 +1,145 @@
+/***************************************************************************
+  qgshelp.cpp
+  --------------------------------------
+  Date                 : December 2016
+  Copyright            : (C) 2016 by Alexander Bruy
+  Email                : alexander dot bruy at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgshelp.h"
+
+#include <QSettings>
+#include <QLocale>
+#include <QUrl>
+#include <QFileInfo>
+#include <QTcpSocket>
+#include <QDesktopServices>
+
+#include "qgis.h"
+#include "qgsapplication.h"
+
+QgsHelp *QgsHelp::sHelp = nullptr; // Singleton instance
```

Comment:
I'm afraid, in this case we will get circular dependency, as `QgsHelp` includes `qgsapplication.h` and uses `QgsApplication::pkgDataPath()`. I will look if we can avoid it.